src/Cake.Common/Tools/OpenCover/OpenCoverRunner.cs

```diff
@@ -141,7 +141,10 @@ public sealed class OpenCoverRunner : Tool<OpenCoverSettings>
                 builder.Append("-mergeoutput");
             }
 
-            builder.AppendSwitch("-register", ":", settings.Register);
+            if (settings.Register != null)
+            {
+                builder.AppendSwitch("-register", string.Empty, settings.Register.ToString());
```

Comment:
I personally would prefer to have the quote in here as second argument instead of the `ToString` in `Register`

Reply:
@Roemer Having the colon here leads to `-register:` for `OpenCoverRegisterOption.Admin` which in turn is not valid. So the alternative would be something like 
```
var registerValue = settings.Register.toString();
if(string.isNullOrEmtpy(registerValue){
   builder.AppendSwitch("-register");
} else {
  builder.AppendSwitch("-register", ":", registerValue);
}
```

which imho is much less readble. What do you think?

(or is it possible to have `AppendSwitch` omit the separator if the last parameter is empty?)

Reply:
Ah I see. Maybe add this information as a comment above this line so nobody would change and break that in the future.